packages/jupyter-ai/src/components/chat-input.tsx

```diff
@@ -26,8 +39,122 @@ type ChatInputProps = {
   sx?: SxProps<Theme>;
 };
 
+type SlashCommandOption = {
+  id: string;
+  label: string;
+  description: string;
+};
+
+/**
+ * List of icons per slash command, shown in the autocomplete popup.
+ *
+ * This list of icons should eventually be made configurable. However, it is
+ * unclear whether custom icons should be defined within a Lumino plugin (in the
+ * frontend) or served from a static server route (in the backend).
+ */
+const DEFAULT_SLASH_COMMAND_ICONS: Record<string, JSX.Element> = {
+  ask: <FindInPage />,
+  clear: <HideSource />,
+  export: <Download />,
+  generate: <MenuBook />,
+  help: <Help />,
+  learn: <School />,
+  unknown: <MoreHoriz />
+};
+
+/**
+ * Renders an option shown in the slash command autocomplete.
+ */
+function renderSlashCommandOption(
+  optionProps: React.HTMLAttributes<HTMLLIElement>,
+  option: SlashCommandOption
+): JSX.Element {
+  const icon =
+    option.id in DEFAULT_SLASH_COMMAND_ICONS
+      ? DEFAULT_SLASH_COMMAND_ICONS[option.id]
+      : DEFAULT_SLASH_COMMAND_ICONS.unknown;
+
+  return (
+    <li {...optionProps}>
+      <Box sx={{ lineHeight: 0, marginRight: 4, opacity: 0.618 }}>{icon}</Box>
+      <Box sx={{ flexGrow: 1 }}>
+        <Typography
+          component="span"
+          sx={{
+            fontSize: 'var(--jp-ui-font-size1)'
+          }}
+        >
+          {option.label}
+        </Typography>
+        <Typography
+          component="span"
+          sx={{ opacity: 0.618, fontSize: 'var(--jp-ui-font-size0)' }}
+        >
+          {' — ' + option.description}
```

Comment:
Minor: Should we avoid displaying this if `option.description` is blank, or contains only whitespace? This would prevent seeing "**/foo** — " if a custom command had a blank description.